modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java

```diff
@@ -365,27 +373,36 @@ public Response installApplication(@HeaderParam("Accept") String acceptHeader,
 		Message message = new Message();
 
 		try {
-			ConfigOperation configOperation = new ConfigOperation();
-			configOperation.setCode(AndroidConstants.OperationCodes.INSTALL_APPLICATION);
-			configOperation.setType(Operation.Type.CONFIG);
+			InstallApplication installApplication = installApplicationBeanWrapper.getOperation();
 
-			return AndroidAPIUtils.getOperationResponse(applicationBean.getDeviceList(),
-			                                            configOperation, message, responseMediaType);
+			if (installApplication == null) {
+				throw new OperationManagementException("Install application bean is empty");
+			}
+
+			ProfileOperation operation = new ProfileOperation();
+			operation.setCode(AndroidConstants.OperationCodes.INSTALL_APPLICATION);
+			operation.setType(Operation.Type.PROFILE);
+			operation.setPayLoad(installApplication.toJSON());
+
+			return AndroidAPIUtils.getOperationResponse(installApplicationBeanWrapper.getDeviceIDs(),
+					operation, message, responseMediaType);
 		} catch (OperationManagementException e) {
 			message.setResponseMessage("Issue in retrieving operation management service instance");
 			throw new AndroidOperationException(message, responseMediaType);
 		} catch (DeviceManagementException e) {
 			message.setResponseMessage("Issue in retrieving device management service instance");
 			throw new AndroidOperationException(message, responseMediaType);
+		} catch (OperationConfigurationException e) {
+			message.setResponseMessage("Issue in setting up payload in operation");
+			throw  new AndroidOperationException(message, responseMediaType);
 		}
 	}
 
-	//TODO : Need to properly construct ConfigOperation
 	@POST
 	@Produces(MediaType.APPLICATION_JSON)
 	@Path("uninstall-application")
```

Comment:
better if the path name can be changed to a noun instead of a verb (i.e. install-application) - as REST APIs normally represent resources